index.js

```diff
@@ -1,284 +1,161 @@
-    const d = Buffer.from(b64part, 'base64')
-    return Buffer.concat([SIGNATURE_TF, d])
-  },
 
-  string(str) {
-    const d = Buffer.from(str, 'utf8')
-    return Buffer.concat([STRING_TF, d])
-  },
+    /* fallback */
+    return encoder.string(input)
+  }
 
-  boolean(bool) {
-    const d = bool ? BOOL_TRUE : BOOL_FALSE
-    return Buffer.concat([BOOL_TF, d])
-  },
-}
+  if (typeof input === 'boolean') return encoder.boolean(input)
+  if (input === null) return encoder.nil()
 
-function encode(input) {
+  /* recursions */
   if (Array.isArray(input)) {
     return input.map((x) => {
       const y = encode(x)
-      if (y === undefined) return NIL_TFD
-      else return y
+      return y === undefined ? encoder.nil() : y
     })
   }
-  if (input === undefined) {
-    return undefined
-  } else if (input === null) {
-    return NIL_TFD
-  } else if (typeof input === 'object' && !Buffer.isBuffer(input)) {
+  if (typeof input === 'object') {
+    // we have already checked it's not a Buffer/null/Array
     const output = {}
-    for (let key in input) {
-      const x = input[key]
-      const y = encode(x)
+    for (const key in input) {
+      const y = encode(input[key])
       if (y !== undefined) output[key] = y
     }
     return output
-  } else if (typeof input === 'string') {
-    if (isFeedType(input)) return encoder.feed(input)
-    else if (isMsgType(input)) return encoder.message(input)
-    else if (isBlobType(input)) return encoder.blob(input)
-    else if (input.match(/\.sig\.[a-zA-Z0-9]+$/))
-      return encoder.signature(input)
-    else if (input.match(/\.box\d*$/)) return encoder.box(input)
-    else return encoder.string(input)
-  } else if (typeof input === 'boolean') {
-    return encoder.boolean(input)
-  } else {
-    if (!Number.isInteger(input) && !Buffer.isBuffer(input))
-      console.warn('not encoding unknown value', input)
-    // FIXME: more checks, including floats!
-    return input
   }
+
+  throw new Error('cannot encode input ' + input)
 }
 
 const decoder = {
-  feed(buf) {
-    const tf = buf.slice(0, 2)
-    const d = buf.slice(2)
-
-    let feedExtension
-    if (tf.equals(CLASSIC_FEED_TF)) feedExtension = '.ed25519'
-    else if (tf.equals(GABBYGR_FEED_TF)) feedExtension = '.ggfeed-v1'
-    else if (tf.equals(BENDYBT_FEED_TF)) feedExtension = '.bbfeed-v1'
-    else throw new Error('Unknown feed: ' + buf)
-
-    const b64part = d.toString('base64')
-    return '@' + b64part + feedExtension
-  },
-
-  message(buf) {
-    const tf = buf.slice(0, 2)
-    const d = buf.slice(2)
-
-    let msgExtension
-    if (tf.equals(CLASSIC_MSG_TF)) msgExtension = '.sha256'
-    else if (tf.equals(GABBYGR_MSG_TF)) msgExtension = '.ggmsg-v1'
-    else if (tf.equals(BENDYBT_MSG_TF)) msgExtension = '.bbmsg-v1'
-    else throw new Error('Unknown msg: ' + buf)
-
-    const b64part = d.toString('base64')
-    return '%' + b64part + msgExtension
+  sigilSuffix(input, type, format) {
+    return [
```

Comment:
Can you bring back `const d = input.slice(2)` which was a pattern used everywhere here?

Reply:
done